Formula/octave.rb

```diff
@@ -82,6 +82,9 @@ def install
               /%OCTAVE_CONF_OCT(AVE)?_LINK_(DEPS|OPTS)%/,
               '""'
 
+    # SUNDIALS 6.4.0 and later needs C++14 for C++ based features
+    ENV.append "CXXFLAGS", "-std=c++14"
```

Comment:
It would be better to set it to `-std=gnu++14` because Octave uses some GNU extensions (and only falls back to some inferior workarounds if they aren't available).

Also, setting this only in `CXXFLAGS` might cause some preprocessor tests to get different results during configuration than what the compiler would see when building.
It might be better to add this flag to `CXX`.

Reply:
@cho-m: I hope it is ok to comment on a closed PR. But this should definitely be fixed in the Octave build rules.
I.e.:
``` suggestion
    ENV.append "CXX", "-std=gnu++14"
```